parser/prism/Translator.cc

```diff
@@ -3671,17 +3696,24 @@ NodeVec Translator::translateArguments(pm_arguments_node *argsNode, pm_node *blo
 ast::ExpressionPtr Translator::desugarArray(core::LocOffsets location, absl::Span<pm_node_t *> prismElements,
                                             ast::Array::ENTRY_store elements) {
     auto locZeroLen = location.copyWithZeroLength();
+    auto calledFromCallNode = location.empty();
 
     ast::Array::ENTRY_store elems;
     elems.reserve(elements.size());
 
     ExpressionPtr lastMerge;
-    ENFORCE(elements.size() == prismElements.size());
-    for (int i = 0; i < elements.size(); i++) {
-        auto *node = prismElements[i];
-        auto &stat = elements[i];
+    ENFORCE(elements.size() <= prismElements.size());
```

Comment:
This gets more hairy now. I wanted to base the branch decisions here on the Prism nodes, but those are no longer 1:1 corresponding with elements of the sorbet elements (some of those might be removed). So I keep two separate indices to match the elements up

This will all simplify out once the whitequark nodes are gone